testutil/testproxy/relayerproxy.go

```diff
@@ -151,10 +151,14 @@ func WithServicesConfigMap(
 				server.Handler = http.HandlerFunc(func(w http.ResponseWriter, _ *http.Request) {
 					sendJSONRPCResponse(test.t, w)
 				})
-				go func() { server.ListenAndServe() }()
+				go func() {
+					err := server.ListenAndServe()
+					require.NoError(test.t, err)
```

Comment:
This seems to be producing new `bind: address already in use` errors :thinking: :

![image](https://github.com/pokt-network/poktroll/assets/600733/23ac95cc-d614-47c9-aff5-8569e4f2eb2a)

Reply:
There were a couple of issues:
- We tried to create new proxies with the same config while already having one running for the duration of whole test.
- We did not cleanup the servers properly. 

The fix incoming! :)